### PR TITLE
Wiring the Update service together

### PR DESCRIPTION
**Pull request by rtyler**
SEVERE WORK IN PROGRESS :smile: 

I don't want to merge this until the pieces are a bit more cleanly wired.

### CODE REVIEW

File: services/Dockerfile [hunk 1]
```diff
@@ -24,7 +24,5 @@ COPY public ${APP_DIR}/public
 EXPOSE 3030
 
 COPY wait-for-postgres.sh /wait-for-postgres.sh
-RUN chmod a+x /wait-for-postgres.sh
-
-RUN apk add --update-cache postgresql-client
-CMD ["/wait-for-postgres.sh", "db", "5432", "/usr/local/bin/npm", "run", "start"]
+RUN apk add --update-cache postgresql-client && chmod a+x /wait-for-postgres.sh
+CMD npm run start
```
[the remark this answers is not in the document]
Comment (rtyler) [reply services/Dockerfile]: `wait-for-postgres.sh` is still required for the `docker-compose.yml` in the root directory

File: client/src/client.js [hunk 2]
```diff
@@ -21,18 +25,58 @@ class Client {
...
+    this.status.authenticate(this.reg.uuid, token);
+    this.update.authenticate(this.reg.uuid, token);
+
+    this.statusAndUpdate();
+
+    cron.runHourly('post-status', () => {
+      this.status.create();
+    });
+    cron.runDaily('check-for-updates', () => {
+      this.statusAndUpdate();
```
[the remark this answers is not in the document]
Comment (rtyler) [reply client/src/client.js]: We will still have a push channel ("pusher")

File: client/src/client.js [hunk 3]
```diff
@@ -21,18 +25,58 @@ class Client {
...
     this.reg = new Registration(this.app);
     this.status = new Status(this.app);
+    this.update = new Update(this.app);
     this.errorTelemetry = new ErrorTelemetry(this.app);
   }
 
+  statusAndUpdate() {
+    this.status.create().then(() => {
+      logger.info('Status created, checking for updates');
+      this.update.query().then((ups) => {
```
[the remark this answers is not in the document]
Comment (rtyler) [reply client/src/client.js]: Updates :smiling_imp:

File: client/src/client.js [hunk 4]
```diff
@@ -21,18 +25,58 @@ class Client {
...
+
+          if (ups.core.url) {
+            Downloader.download(ups.core.url, dir).then((stream) => {
+              logger.info('Core downloaded', stream.path);
+            });
+          }
+
+          ups.plugins.updates.forEach((plugin) => {
+            logger.info('Fetching ', plugin.url);
+            Downloader.download(plugin.url, [dir, 'plugins'].join(path.sep)).then((stream) => {
```
[the remark this answers is not in the document]
Comment (rtyler) [reply client/src/client.js]: This does need to change but I consider it outside of the scope of this change. This is related to the discussion we had last week about how we need to change symbolic links around as part of the update lifecycle.

This PR is largely getting the client/server interaction in place to handle the updates in a manner similar to our current aria-hard-coding.

We still have more work to do with integration of updates and snapshots

File: client/src/client.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (batmat) [reply client/src/client.js]: I thought it was 'Oops [I did it again]' :hear_no_evil:

File: client/src/client.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (batmat) [reply client/src/client.js]: Fine by me to see it as a first stab indeed and improve by iterations.

**Comment by rtyler** [comment]
I have changed the base branch and will merge this so @batmat have a place to land new commits while we sort out the broken integration tests

**Pull request merged**